csp/tests/test_utils.py

```diff
@@ -211,3 +211,8 @@ def test_upgrade_insecure_requests():
 def test_block_all_mixed_content():
     policy = build_policy()
     policy_eq("default-src 'self'; block-all-mixed-content", policy)
+
+
+def test_nonce():
+    policy = build_policy(nonce='abc123')
+    policy_eq("default-src 'self'; nonce-abc123", policy)
```

Comment:
What about having a separate setting like:

```python
CSP_INCLUDE_NONCE_IN = ["script-src", "style-src"]
```

That way, you could easy specify which sources it's added to, or you could just set that list to `["default-src"]` (which should probably be the default).

This would also negate the need for the separate `CSP_DISABLE_NONCE` setting.  Because if `CSP_INCLUDE_NONCE_IN` is set to empty, then you know you don't need to generate a nonce.